server/src/routes/donatedItemRoutes.ts

```diff
@@ -55,17 +56,64 @@ router.post('/', donatedItemValidator, async (req: Request, res: Response) => {
 // GET /donatedItem - Fetch all donated items
 router.get('/', async (req: Request, res: Response) => {
     try {
-        const items = await prisma.donatedItem.findMany({
+    const donatedItem = await prisma.donatedItem.findMany({
+        include: {
+            donor: true, // Include all donor details
+            program: true, // Include all program details
+            statuses: { 
+                orderBy: {
+                    dateModified: 'asc' // Ensure they are ordered chronologically
```

Comment:
There seems to be an odd interaction where it sometimes gives updates that it should not